src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -15,80 +15,63 @@ public class HeapElement
         private const int kInitialSize = 5;
         private const double kReallocFactor = 0.5;
 
-        private int AllocSize { get; set; }
+        protected Heap heap;
+        private int allocated;
+        private StackValue[] items;
+
```

Comment:
Just a minor point, but it doesn't seem useful to have a dedicated data member `allocated` if we're able to infer it from `items.Length`. having one more member might make it more error-prone I tend to think.